src/common/config.cc

```diff
@@ -1236,21 +1236,45 @@ bool md_config_t::expand_meta(std::string &origval,
 }
 
 void md_config_t::diff(
+  const md_config_t *other,
+  map<string, pair<string, string> > *diff,
+  set<string> *unknown) 
+{
+  diff_helper(other, diff, unknown, "");
+}
+void md_config_t::diff(
+  const md_config_t *other,
+  map<string, pair<string, string> > *diff,
+  set<string> *unknown, const string& ceph_setting = string("")) 
+{
+  diff_helper(other, diff, unknown, ceph_setting);
+}
+
+void md_config_t::diff_helper(
     const md_config_t *other,
     map<string,pair<string,string> > *diff,
-    set<string> *unknown)
+    set<string> *unknown, const string& ceph_setting=string(""))
 {
   Mutex::Locker l(lock);
 
+  bool ceph_param_search = (!ceph_setting.empty()) ? true : false;
```

Comment:
```c++
bool ceph_param_search = !ceph_setting.empty();
```
would suffice. and IMHO, no need to introduce `ceph_param_search` just for checking if `ceph_setting` is empty or has the default value.